Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/core/processor/DccReissuanceProcessor.kt

```diff
@@ -0,0 +1,93 @@
+package de.rki.coronawarnapp.dccreissuance.core.processor
+
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.Certificate
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.CertificateReissuance
+import de.rki.coronawarnapp.covidcertificate.common.certificate.CertificateProvider
+import de.rki.coronawarnapp.covidcertificate.common.qrcode.DccQrCode
+import de.rki.coronawarnapp.dccreissuance.core.error.DccReissuanceException
+import de.rki.coronawarnapp.dccreissuance.core.server.DccReissuanceServer
+import de.rki.coronawarnapp.dccreissuance.core.server.data.DccReissuanceResponse
+import de.rki.coronawarnapp.qrcode.handler.DccQrCodeHandler
+import de.rki.coronawarnapp.qrcode.scanner.QrCodeValidator
+import de.rki.coronawarnapp.reyclebin.covidcertificate.RecycledCertificatesProvider
+import de.rki.coronawarnapp.tag
+import kotlinx.coroutines.flow.first
+import timber.log.Timber
+import javax.inject.Inject
+
+class DccReissuanceProcessor @Inject constructor(
+    private val dccReissuanceServer: DccReissuanceServer,
+    private val certificateProvider: CertificateProvider,
+    private val recycledCertificatesProvider: RecycledCertificatesProvider,
+    private val qeCodeValidator: QrCodeValidator,
+    private val dccQrCodeHandler: DccQrCodeHandler
+) {
+
+    @Throws(DccReissuanceException::class)
+    suspend fun requestDccReissuance(dccReissuanceDescriptor: CertificateReissuance) {
+        Timber.tag(TAG).d("requestDccReissuance(dccReissuanceDescriptor=%s)", dccReissuanceDescriptor)
+
+        // 1. Call API
+        val dccReissuanceResponse = dccReissuanceDescriptor.callApi()
+
+        // 2. Add DCC to Wallet
+        dccReissuanceResponse.addDccToWallet()
+
+        // 3. Move previous DCC to bin
+        dccReissuanceDescriptor.certificateToReissue.recycleDcc()
```

Comment:
I don't think that step 2,3 should be of a concern here, just provide the response and whatever needs this response should do that + The repository have new functions that takes old and new Certificates do do the replacement in on operation, what you are doing here will trigger the dccWallet calculation twice

Reply:
I can remove it and let the caller implement the steps.
I totally missed the new replace funtions..

Reply:
That would better IMHO